# **Cleaning Titanic dataset as my first Data analize project**
****OBJECTIVE: clean the messy data and get insight from the clean data ***** 

In [1]:
import pandas as pd 

data = pd.read_csv('/kaggle/input/datasets/yasserh/titanic-dataset/Titanic-Dataset.csv') # for load csv data .
df=data.copy() # make the copy of original data because if i lost teh data i will get it back .
df.head() # for understand , what kind of data i have 


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


# Get the info of data

1. By using (.info) i get the all info about data like : missing value , how many rows and column the data has ect.
2. By using (.isnull().sum()) i get the rows that have missing values .

In [2]:
# data cleaning
 
df.info() # for get the all info about data 
df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

# Clean the data 

1. Column names are inconsistent , so i convert them into lower case and the space will fill by _ .
2. IN this data 714 cell of age column are null , so i use (.groupby) in pclass to get the median of every pclass then fill the age cell by the     median of thier pclass.
3. In cabin column 77% of data is null so i am  delete the cabin column,  because fill the null cells with 0 or average is not making sense.
4. Embarked means here "The port where passenger boarded the titanic , embarks column has only two cell of missing data and teh "s" 644 passenger are boarded from "s" so i fill the missing cell with "s".
5. The dataset has no duplicate value.

# 

In [3]:
df.columns=(df.columns.str.lower().str.strip().str.replace(" ","_"))
df["age"]= df["age"].fillna(df.groupby("pclass")["age"].transform("median"))
df.drop("cabin",axis=1,inplace=True)
df["embarked"]=df["embarked"].fillna(df["embarked"].mode()[0])
df.duplicated().sum()

np.int64(0)

#  Insight from the data .
1. People from "q" (embarked) pay more fare.
2. 342 people are survived in Titanic crash.549 people are died.
4. 74% female and 18% male are survived
5. Most of the 1st class people are survived.
6. Young people are survived most.
7. People who pay more fare are survived.
8. Most people are travel in lower class only 216 people rae travel in high class.
9. People who boarded from "c" are survived most.

In [4]:

# df.describe()
df.groupby("embarked")["fare"].mean()
df["survived"].value_counts()
df.groupby("sex")["survived"].mean()
df.groupby("pclass")["survived"].mean()
df.groupby("survived")["age"].mean()
df.groupby("survived")["fare"].mean()
df["pclass"].value_counts()
df.groupby("embarked")["survived"].mean()

    



embarked
C    0.553571
Q    0.389610
S    0.339009
Name: survived, dtype: float64

# File save
**clean file is save as"clean titanic_data" in csv formate**

In [5]:
import os

os.listdir("/kaggle/working")

df.to_csv(
    "cleaned titanic_data",
    index=False
)